app/blog/posts/the-modern-frontend-workflow.md

```diff
@@ -0,0 +1,88 @@
+
+## The modern frontend workflow
+
+It should allow frontend developers to create great resuable building blocks that are not tied to any backend language, but can be used by any.
+
+It should allow creating high quality responsive designs without testing manually in the browser to see if something works.
+
+It should allow quick iterations and easily sharing of prototypes without extra or wasted work.
+
+### Webcomponents
+
+Webcomponents are the most important aspect of the modern frontend workflow. It is a huge leap because it allows frontenders to create reuseable building blocks that can be used anywhere. Webcomponents can be either big or small, but it allows encapsulation of HTML, CSS and JS that can be either a complete app or a small vital replacement component for a `<input>` field.
+
+It makes it possible for backend developers to use new webcomponents in their layout files as easily as html tags. Frontend developers can make new html tags and backend developers can consume them. Everything can be kept small and lean.
+
+https://developers.google.com/web/fundamentals/web-components/customelements
+
+### Storybook
+
+A great website has great error handling! With storybook you can show how that error handling looks like with a predefined story. As a frontend developer you should not have to sit and fill out forms endlessly to test how the error handling flow looks like.
+
+Make a story for each: a story for a succesful form, a story for a error in a form, a story for clicking the submit, a story for a empty form. Now more time can be spent perfecting those often hidden states of your application.
+
+https://storybook.js.org/
+
+### Components with React.js
+
+React.js is the most popular framework to create interactive experiences in the browser and the reason why is the ability to create small components that you can easily compose into larger ones. Webcomponents is the the standard, but being a standard it is both more low level and moves slower. They both benefit, but React.js benefits right now by being the most popular and therefore having the most atttention in terms of tools and extensions built for it.
+
+Building something for others to consume, Webcomponents. Building something to consume yourself, choose React.js and you are sure to be developing minimal components that can be adapted easily.
+
+https://reactjs.org/docs/components-and-props.html
+
+### CSS modules
+
+All visual, all colors and styling are made with CSS, but CSS are dificult because of the fact it is Cascading as the name says. CSS Modules allows you to focus on the styling for one component at a time, allowing less concerns about how it affect the whole app and avoding side effects.
```

Comment:
```suggestion
All visual, all colors and styling are made with CSS, but CSS is difficult because of the fact that it is Cascading, as the name says. CSS Modules allows you to focus on the styling for one component at a time, allowing less concerns about how it will affect the whole app and avoiding side effects.
```